common/src/main/java/com/microsoft/identity/common/exception/CommonCoreBaseException.java

```diff
@@ -0,0 +1,149 @@
+    }
+
+    /**
+     * Initiates the detailed error code.
+     *
+     * @param errorCode The error code contained in the exception.
+     */
+    public CommonCoreBaseException(final String errorCode) {
+        mErrorCode = errorCode;
+    }
+
+    /**
+     * Initiates the {@link CommonCoreBaseException} with error code and error message.
+     *
+     * @param errorCode    The error code contained in the exception.
+     * @param errorMessage The error message contained in the exception.
+     */
+    public CommonCoreBaseException(final String errorCode, final String errorMessage) {
+        super(errorMessage);
+        mErrorCode = errorCode;
+    }
+
+    /**
+     * Initiates the {@link CommonCoreBaseException} with error code, error message and throwable.
+     *
+     * @param errorCode    The error code contained in the exception.
+     * @param errorMessage The error message contained in the exception.
+     * @param throwable    The {@link Throwable} contains the cause for the exception.
+     */
+    public CommonCoreBaseException(final String errorCode, final String errorMessage,
+                  final Throwable throwable) {
+        super(errorMessage, throwable);
+        mErrorCode = errorCode;
+    }
+
+    /**
+     * @return The error code for the exception, could be null. {@link CommonCoreBaseException} is the top level base exception, for the
+     * constants value of all the error code.
+     */
+    public String getErrorCode() {
+        return mErrorCode;
+    }
+
+    /**
+     * {@inheritDoc}
+     * Return the detailed description explaining why the exception is returned back.
+     */
+    @Override
+    public String getMessage() {
+        if (!StringExtensions.isNullOrBlank(super.getMessage())) {
+            return super.getMessage();
+        }
+
+        return "";
+    }
+
+    protected int mHttpStatusCode;
+
+    protected HashMap<String, String> mHttpResponseBody = null;
+
+    protected HashMap<String, List<String>> mHttpResponseHeaders = null;
+
+    /**
+     * When {@link java.net.SocketTimeoutException} is thrown, no status code will be caught. Will use 0 instead.
+     */
+    static final int DEFAULT_STATUS_CODE = 0;
+    /**
+     * @return The http status code for the request sent to the service.
+     */
+    public int getHttpStatusCode() {
+        return mHttpStatusCode;
+    }
+
+    /**
+     * Gets the response body that may be returned by the service.
+     *
+     * @return response body map, null if not initialized.
+     */
+    public HashMap<String, String> getHttpResponseBody() {
+        return mHttpResponseBody;
+    }
+
+    /**
+     * Get the response headers that indicated an error.
+     *
+     * @return The response headers for the network call, null if not initialized.
+     */
+    public HashMap<String, List<String>> getHttpResponseHeaders() {
+        return mHttpResponseHeaders;
+    }
+
+    public void setHttpResponse(HttpWebResponse response) {
```

Comment:
There isn't any constructor taking HttpWebResponse as a parameter in CommonCoreBaseException. Does it mean we need to call setHttpResponse explicitly in other line when creating the CommonCoreBaseException object with http response info?

Reply:
Yes, The current PR was just to review the exception layout, I have not plumbed in the function calls yet.